pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordReaderUtils.java

```diff
@@ -50,7 +49,7 @@ public static BufferedInputStream getBufferedInputStream(File dataFile)
 
   public static InputStream getInputStream(File dataFile)
       throws IOException {
-    if (dataFile.getName().endsWith(GZIP_FILE_EXTENSION)) {
+    if (GzipUtils.isGZipped(dataFile)) {
```

Comment:
Suggest directly trying to open a GZIPInputStream, which will do all the necessary validations (besides the magic short check which might not be enough)
```suggestion
    FileInputStream fileInputStream = new FileInputStream(dataFile);
    try {
      return new GZIPInputStream(fileInputStream);
    } catch (Exception e) {
      // NOTE: Cannot reuse the input stream because it might already be read
      fileInputStream.close();
      return new FileInputStream(dataFile);
    }
```

Reply:
The suggestion seems to rely on exception being thrown as part of logic to determine if this is a Gzipped file or not, which doesn't seem like the best idea to me. Also, could you elaborate on what is the issue with checking magic header?

Reply:
+1, I would check the header rather than assume then throw too

Reply:
Having seen the implementation, which essentially does just that, I have second thoughts

Reply:
My concern is mostly on the size of the header, which is a short that can potentially cause false positive. Reading `GZIPInputStream.readHeader()`, it actually performs much more checks including CRC. If both of you prefer the header check only, then I'm fine with it

Reply:
Yeah, this one is a borderline for me too, so willing to make an `exception` (pun intended).